src/test/java/com/sigma/cinematicketpro/controller/MovieControllerTest.java

```diff
@@ -0,0 +1,168 @@
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.RequestBuilder;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+
+import static com.sigma.cinematicketpro.TestUtils.getObjectMapper;
+import static com.sigma.cinematicketpro.TestUtils.mapMvcResultIntoObject;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@WebMvcTest(controllers = MovieController.class)
+@ExtendWith(MockitoExtension.class)
+class MovieControllerTest {
+    protected static final List<Movie> movieList = TestUtils.getValidMovies();
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @MockBean
+    private MovieService movieService;
+    @MockBean
+    private MovieMapper movieMapper;
+
+    @Test
+    void shouldReturnListOfMovies_WhenGetAllMovies() throws Exception {
+        List<Movie> expectedMovieList = movieList;
+        when(movieService.getAllMovies()).thenReturn(expectedMovieList);
+
+        RequestBuilder request = MockMvcRequestBuilders.get("/movies");
+
+        MvcResult response = mockMvc.perform(request)
+                .andExpect(content().contentType(MediaType.APPLICATION_JSON))
+                .andDo(print())
+                .andReturn();
+
+        List<Movie> actualMovieList = mapMvcResultIntoObject(response, new TypeReference<>() {
+        });
+
+        assertThat(expectedMovieList).isEqualTo(actualMovieList);
+        verify(movieService, times(1)).getAllMovies();
+    }
+
+    @Test
+    void shouldThrowResourceNotFoundException_WhenGetMovieByIdAndMovieIdDoesntExist() throws Exception {
+        String movieId = "movieId";
+
+        when(movieService.getMovieById(movieId)).thenThrow(ResourceNotFoundException.class);
+
+        RequestBuilder request = MockMvcRequestBuilders.get("/movies/{id}", movieId);
+
+        mockMvc.perform(request)
+                .andExpect(status().isNotFound());
+
+        verify(movieService, times(1)).getMovieById(movieId);
+    }
+
+    @Test
+    void shouldReturnMovie_WhenGetMovieById() throws Exception {
+        Movie expectedMovie = movieList.get(0);
+        String movieId = expectedMovie.getId();
+
+        when(movieService.getMovieById(movieId)).thenReturn(expectedMovie);
+
+        RequestBuilder request = MockMvcRequestBuilders.get("/movies/{id}", movieId);
+
+        MvcResult response = mockMvc.perform(request)
+                .andExpect(content().contentType(MediaType.APPLICATION_JSON))
+                .andDo(print())
```

Comment:
why do you need this? this is regards all occurences

Reply:
It's basically to print the request/response in the console. Added a check for the response code here to make it more meaningful.